apps/api/src/app/workflows-v2/util/build-variables.ts

```diff
@@ -13,9 +14,11 @@ export function buildVariables(
 ): TemplateVariables {
   let variableControlValue = controlValue;
 
-  if (isStringTipTapNode(variableControlValue)) {
+  if (isStringifiedMailyJSONContent(variableControlValue)) {
     try {
-      variableControlValue = new HydrateEmailSchemaUseCase().execute({ emailEditor: variableControlValue });
+      variableControlValue = new WrapMailyInLiquidUseCase().execute({
+        emailEditor: variableControlValue,
+      });
```

Comment:
I feel like this bit of the code should be extracted to the places where this function `buildVariables` is used and `wrapMailyInLiquidUseCase` usecase injected there like normally we do. There are only two places.

Reply:
I generally agree, it doesn't need to be an usecase necessarily, but I feel like usecase is more contained - its in one folder, it has types, command, clear structure etc. Utility functions tend to just be hanging somewhere with vague names:

![image](https://github.com/user-attachments/assets/e35100d7-b776-4aff-aa9a-aaeacaeafded)